sdk/core/README.md

```diff
@@ -1,12 +1,3 @@
-# Azure SDK for Rust - Core crate
+# azure_core crate
```

Comment:
I don't really love these changes to the readme...

Perhaps reverting this and then opening a new PR where we can discuss the needs of a README and how we can address them?